rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -216,7 +216,9 @@ class Publisher : public PublisherBase
   virtual void
   publish(std::unique_ptr<MessageT, MessageDeleter> & msg)
   {
-    this->do_inter_process_publish(msg.get());
+    if (!use_intra_process_ || get_subscription_count() > get_intra_process_subscription_count()) {
```

Comment:
If the QoS is not volatile, then it should always do inter process.

Reply:
If QoS is non volatile, use_intra_process_ flag=false, as setup intraprocess doesn't set it to true. It just return without setting-up the intraprocess comm.

Reply:
You're right, I was thinking that was the flag the user set.

----

This logic could use a comment, I'd even support making an intermediate bool to make it clearer why you're comparing the count's. For example:

```suggestion
    bool intra_process_is_enabled = use_intra_process_;  // might consider renaming this variable entirely
    bool inter_process_subscriptions_exist = get_subscription_count() > get_intra_process_subscription_count();
    if (!intra_process_is_enabled || inter_process_subscriptions_exist) {
```